### ISSUE TITLE
Safe mode

### ISSUE THREAD

**Issue by tkurki** [SignalK/signalk-server-node#792]
We should have a way to (automatically) start in "safe mode" so that users do not end up in crash loops where something is repeatedly crashing and one can not get the admin UI up to disable the crashing components.

**Comment by tkurki** [comment]
Implementation idea: 
- create a marker file in $TMP very early during startup
- after the server is fully started / some time has passed remove the marker file
- check the marker's presence before creating it: if the marker is there the server probably crashed before removing it and we can assume safe mode is needed

Make safe mode prominent
- show in admin ui
- send notification deltas

Provide a way to disable the safe mode in the admin ui

**Comment by wouterdebie** [comment]
Is this mainly because plugins crash the server? Are there any scenarios you have that put the server in a bad state? A safe mode seems a bit like a workaround for a larger issue.

**Comment by tkurki** [comment]
Just this summer a plugin that I was testing crashed the server with excessive memory usage and it would not work properly. Especially the plugin system is vulnerable to runaway code that hogs memory or cpu.

Architecturally we can solve this by separating plugins to a separate process, but that adds complexity to the system.

Also it is not just plugins but bugs in server proper that can cause problems.

Testing everything exhaustively is not feasible with the effort available, so safeguards like the one proposed here would be valuable in making the system more resilient.

**Comment by wouterdebie** [comment]
I hear you. I guess an easier way of disabling plugins outside of the server (or when the server is not running) would be helpful too.

**Comment by tkurki** [comment]
If the server is run under systemd modifying startup params or env is not very straightforward. Maybe some _disable-plugins_ marker file? Renaming the plugin config dir will do the trick, except for olygins that are enabled by default.

**Comment by wouterdebie** [comment]
Yes, I was thinking something like that. Put  something in place so that the next time the server starts plugins (or a specific plugin) isn't loaded.
